src/coreclr/jit/gentree.cpp

```diff
@@ -19390,6 +19412,23 @@ GenTree* Compiler::gtNewSimdBinOpNode(
             {
                 assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F));
                 intrinsic = NI_AVX512F_And;
+
+                if (varTypeIsIntegral(simdBaseType))
```

Comment:
Do these special checks where we do forwarding have test coverage?

Reply:
Yes, all the JIT changes have coverage and have been reviewed already.